test/testfolderman.cpp

```diff
@@ -24,7 +26,105 @@ class TestFolderMan: public QObject
+        QVERIFY(secondShare);
+        secondShare->permissions.setPermission(OCC::RemotePermissions::IsShared);
+
+        FolderMan *folderman = FolderMan::instance();
+        QCOMPARE(folderman, &_fm);
+        OCC::AccountState *accountState = OCC::AccountManager::instance()->accounts().first().data();
+        const auto folder = folderman->addFolder(accountState, folderDefinition(fakeFolder.localPath()));
+        QVERIFY(folder);
+
+        auto realFolder = FolderMan::instance()->folderForPath(fakeFolder.localPath());
+        QVERIFY(realFolder);
+
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        fakeQnam->setOverride([this, accountState, &fakeFolder](QNetworkAccessManager::Operation op, const QNetworkRequest &req, QIODevice *device) {
+            Q_UNUSED(device);
+            QNetworkReply *reply = nullptr;
+
+            if (op != QNetworkAccessManager::DeleteOperation) {
+                reply = new FakeErrorReply(op, req, this, 405);
+                return reply;
+            }
+
+            if (req.url().path().isEmpty()) {
+                reply = new FakeErrorReply(op, req, this, 404);
+                return reply;
+            }
+
+            const auto filePathRelative = req.url().path().remove(accountState->account()->davPath());
+
+            const auto foundFileInRemoteFolder = fakeFolder.remoteModifier().find(filePathRelative);
+
+            if (filePathRelative.isEmpty() || !foundFileInRemoteFolder) {
+                reply = new FakeErrorReply(op, req, this, 404);
+                return reply;
+            }
+
+           fakeFolder.remoteModifier().remove(filePathRelative);
+           reply = new FakePayloadReply(op, req, {}, nullptr);
+
+           emit incomingShareDeleted();
+           
+           return reply;
+        });
+
+        QSignalSpy incomingShareDeletedSignal(this, &TestFolderMan::incomingShareDeleted);
+
+        // verify first share gets deleted
+        folderman->leaveShare(fakeFolder.localPath() + firstSharePath);
+        QVERIFY(incomingShareDeletedSignal.count() == 1);
+        QVERIFY(!fakeFolder.remoteModifier().find(firstSharePath));
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        // verify no share gets deleted
+        folderman->leaveShare(fakeFolder.localPath() + "A/B/notsharedwithme_B.data");
+        QVERIFY(incomingShareDeletedSignal.count() == 1);
+        QVERIFY(fakeFolder.remoteModifier().find("A/B/sharedwithme_B.data"));
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        // verify second share gets deleted
+        folderman->leaveShare(fakeFolder.localPath() + secondSharePath);
+        QVERIFY(incomingShareDeletedSignal.count() == 2);
```

Comment:
it would be better to use `QCOMPARE` for that as in case of a failure, you clearly see the expected value and the real value
see other lines similar that use `QVERIFY` to compare real and expected values